server/data/restClientMetricsMiddleware.ts

```diff
@@ -33,6 +34,7 @@ function restClientMetricsMiddleware(agent: SuperAgentRequest) {
     req.on('response', (res: Response, err: Error) => {
       res.on('end', () => {
         const responseTime = Date.now() - startTime
+        inMemoryStore.userVersion = res.headers['x-cas-user-version']
```

Comment:
This doesn't seem to be related to client metrics, would it be better to create a new middleware to do this?